main.go

```diff
@@ -35,15 +60,27 @@ func (svc *Service) getProtectedResource(c *gin.Context) {
 		}
 
 		invoice := lndInvoice.PaymentRequest
+		paymentHash, err := lntypes.MakeHash(lndInvoice.RHash)
+		if err != nil {
+			c.Error(err)
+			return
+		}
 
-		macaroonString, err := macaroon.GetMacaroonAsString(lnInvoice.RHash)
+		macaroonString, err := mintMacaroon.GetMacaroonAsString(paymentHash)
 		if err != nil {
 			c.Error(err)
 			return
 		}
 
 		c.Writer.Header().Set("WWW-Authenticate", fmt.Sprintf("LSAT macaroon=%v, invoice=%v", macaroonString, invoice))
 		c.String(http.StatusPaymentRequired, "402 Payment Required")
+		return
+	}
+
+	// Check valid LSAT
+	if authFieldIsPresent && IsValidLSAT {
```

Comment:
I'm pretty sure `authFieldIsPresent` always is `true` here. You could return the protected content on line 42, and then you can get rid of all the `if` blocks.